Week3/homework/3-step3.js

```diff
@@ -1,25 +1,49 @@
-'use strict';
+"use strict";
 
 // use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
-  // add your code here
-  return str;
+  if (num <= 0) return "";
+  else {
+    let result = "";
+    for (let i = 0; i < num; i++) {
+      result += str;
+    }
+    return result;
+  }
 }
 
-console.log('for', repeatStringNumTimesWithFor('abc', 3));
+repeatStringNumTimesWithFor("abc", 3);
+
+console.log("for", repeatStringNumTimesWithFor("abc", 3));
 
 // use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
-  // add your code here
-  return str;
+  if (num <= 0) {
+    return "";
+  } else {
+    let repeatedStr = "";
+    let i = 0;
+    while (i < num) {
+      repeatedStr += str;
+      i++;
+    }
+    return repeatedStr;
+  }
 }
-
-console.log('while', repeatStringNumTimesWithWhile('abc', 3));
+console.log("while", repeatStringNumTimesWithWhile("abc", 3));
 
 // use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
-  // add your code here
-  return str;
+  if (num <= 0) {
```

Comment:
Contrary to the other two cases, here you need to test for `num <= 0`, because a `do-while` loop tests the condition at the end of the loop. Therefore it may execute the loop even if the loop condition does not hold true. Which makes the `do-while` loop rather useless in practice. I never use it myself.